google/cloud/bigquery/client.py

```diff
@@ -27,13 +27,16 @@
 import json
 import math
 import os
+import packaging.version
 import tempfile
 from typing import Any, BinaryIO, Dict, Iterable, Optional, Sequence, Tuple, Union
 import uuid
 import warnings
 
 try:
     import pyarrow
+
+    _PYARROW_VERSION = packaging.version.parse(pyarrow.__version__)
 except ImportError:  # pragma: NO COVER
     pyarrow = None
```

Comment:
Potentially leaves `_PYARROW_VERSION` undefined.  Maybe:
```suggestion
    pyarrow = None
    _PYARROW_VERSION = None
```

Reply:
It's only used in the path where `pyarrow` is definitely available (with a prior check), thus left it out. Although that implicit fact can change...

On the other hand, we are already making `pyarrow` required, thus it probably shouldn't be an issue in the transitional period, but a valid remark nevertheless.